src/compiler/comments.ts

```diff
@@ -68,38 +68,48 @@ namespace ts {
                 emitDetachedComments,
             };
 
-            function getLeadingComments(range: TextRange | Node, getAdditionalRange?: (range: Node) => Node) {
-                let comments = getLeadingCommentsOfPosition(range.pos);
-                if (getAdditionalRange) {
-                    let additionalRange = getAdditionalRange(<Node>range);
-                    while (additionalRange) {
-                        comments = concatenate(
-                            getLeadingCommentsOfPosition(additionalRange.pos),
-                            comments
-                        );
-
-                        additionalRange = getAdditionalRange(additionalRange);
+            function getLeadingComments(range: TextRange | Node, shouldSkipCommentsForNodeCallback?: (node: Node) => boolean) {
+                if (shouldSkipCommentsForNodeCallback && shouldSkipCommentsForNodeCallback(<Node>range)) {

```

Comment:
Doesn't seem like you should necessarily be converting the range to a `Node` unless you're sure you have a `Node`